src/modules/profile/profile.service.ts

```diff
@@ -14,34 +15,31 @@ export class ProfileService {
   ) {}
 
   async findOneProfile(userId: string) {
-    try {
-      const user = await this.userRepository.findOne({ where: { id: userId } });
-      if (!user) {
-        throw new NotFoundException('User not found');
-      }
-
-      const userProfile = await this.userRepository.findOne({
-        where: { id: userId },
-        relations: ['profile'],
-      });
+    if (!isUUID(userId, '4')) {
+      throw new BadRequestException('Invalid user ID');
+    }
 
-      const profile = userProfile.profile;
-      if (!profile) {
-        throw new NotFoundException('Profile not found');
-      }
+    const user = await this.userRepository.findOne({ where: { id: userId } });
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
```

Comment:
There's a global exception called CustomHttpException for handling errors, please make use of it make use of

Reply:
Done